docs/reprocess-report.md

```diff
@@ -0,0 +1,122 @@
+
+Sidenote: having a blockpage at hand is an opportunity to mine blocked URLs showing same blockpage and mine more blockpages, as different ISPs may show different blockpage for the same blocked URL.
+
+Then _human intelligence task_ should be solved to extract a fingerprint for the blockpage. The fingerprint should be added to the set of fingerprints and `openobservatory/pipeline-shovel` should be rolled out before reprocessing of historical data.
+
+If the blockpage is a static one, there is a fast-path alternative to reprocessing: it's possible to update MetaDB directly without actual reprocessing as SHA256 collision is very unlikely and `body_sha256` may be used as a feature _identifying_ the blockpage server (at the current stage of OONI Methodology development). See feature-based fingerprint case for more on the fast-path. Keep in mind that the HTTP Body substring fingerprint is still _derived_ from the body, so avoiding full-dataset reprocessing may lead to false negatives.
+
+Overall steps needed to mark existing & future measurements are:
+
+- pause ongoing ingestion and ensure that there are no `meta_pg` TaskInstances running
+- update `fingerprint` table in [the database schema](https://github.com/ooni/pipeline/blob/065cccdfeb531e93a22d2aacc05ec05e990f99ee/af/oometa/) following [an example](https://github.com/ooni/pipeline/blob/065cccdfeb531e93a22d2aacc05ec05e990f99ee/af/oometa/003-fingerprints.install.sql#L31-L62) and [roll it out](https://github.com/ooni/sysadmin/blob/4defab8e92a2e53e2679a17214162ed058089e7f/ansible/deploy-pipeline-ddl.yml). The fingerprints are stored in the schema to generate `fingerprint_no serial`.
+- create a temporary table having `msm_no` of the measurements matching the fingerprint _with confidence_ according to the _derived_ features existing in database (e.g. `select msm_no from http_request where body_sha256 = '\x833b2fb8887eed1c0d496670148efa8b6a6e65b89f8df42dbd716464e3cf47a6'` for static blockpages)
+- insert those `msm_no` together with matching `fingerprint_no` into `http_request_fp` table as if those were actually ingested by `centrifugation.py`
+- update `anomaly` and `confirmed` flags in `measurement` table for the affected measurement according to the logic codified in `calc_measurement_flags()`
```

Comment:
There is quite some amount of tricky manual work involved in doing this. Do you think there is a way of doing this in a more automated way, but that requires less human input (which may be error prone)?

Reply:
IMHO, it has too many moving parts WRT performance and scope of systems touched to be "easily" automated.
I think, several manual cycles through the checklist with some variety of inputs are needed before trying to automate it.

Reply:
I see. Is there some recommendations for how to go about adding fingerprints to the DB in batch?

We currently have 46 issues open with the label fingerprintdb (https://github.com/ooni/pipeline/issues?q=is%3Aopen+is%3Aissue+label%3Afingerprintdb) and going through all this dance to ingest each of them is going to require a non-trivial amount of time.

I am happy to tradeoff some performance (CPU time or IO time) for less human time.

Reply:
The only recommendation I can give is the same, just treat the checklist as a declarative, not an imperative text, so, perform a batch of actions for every step. You're right being worried, naive non-batch implementation of the checklist for every ticket will take ~11.5 years to converge.